crates/ty_python_semantic/src/types/infer/builder.rs

```diff
@@ -7251,23 +7336,26 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
     /// Note that this method may infer the type of a given argument expression multiple times with
     /// distinct type context. The provided `MultiInferenceState` can be used to dictate multi-inference
     /// behavior.
-    fn infer_all_argument_types(
+    fn infer_all_argument_types<'a, I, F>(
```

Comment:
Same here. This is also a rather big function that we now monomorphize for every possible combination of `I, F`. It would be nice if it could be avoided.